actors/common/src/main/java/org/sunbird/learner/util/OTPUtil.java

```diff
@@ -0,0 +1,104 @@
+package org.sunbird.learner.util;
+
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import com.warrenstrange.googleauth.GoogleAuthenticatorConfig;
+import com.warrenstrange.googleauth.GoogleAuthenticatorKey;
+import com.warrenstrange.googleauth.KeyRepresentation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.background.BackgroundOperations;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.PropertiesCache;
+import org.sunbird.common.request.Request;
+import org.sunbird.notification.sms.provider.ISmsProvider;
+import org.sunbird.notification.utils.SMSFactory;
+
+public final class OTPUtil {
+
+  private static final int MINIMUM_OTP_LENGTH = 6;
+  private static final int SECONDS_IN_MINUTES = 60;
+
+  private OTPUtil() {}
+
+  public static String generateOTP() {
+    String otpSize = PropertiesCache.getInstance().getProperty(JsonKey.SUNBIRD_OTP_LENGTH);

```

Comment:
Use ProjectUtil.getConfigValue as it checks env variable first and then the resource properties.